pkg/divert/weaver/divert.go

```diff
@@ -80,8 +64,8 @@ func (*Driver) Destroy(_ context.Context) error {
 }
 
 func (d *Driver) GetDivertNamespace() string {
```

Comment:
This is to configure the dns search. In the weaver driver, we add a dns search to the diverted namespace:
https://github.com/okteto/okteto/blob/master/cmd/deploy/proxy.go#L476

This means if you call `db`, it will look for a `db` service in the current namespace, and if it doesn't exist, it fallbacks to the diverted namespace.

In Istio, this is already handled by Istio because virtual service hostnames are global, that's why we don't modify the dns search configuration in the Istio driver.